lib/model/CTokenListDataCategorizerBase.cc

```diff
@@ -618,6 +618,45 @@ CTokenListDataCategorizerBase::calculateCategorizationStatus(std::size_t categor
     return model_t::E_CategorizationStatusOk;
 }
 
+std::size_t CTokenListDataCategorizerBase::numMatches(int categoryId) {
+    if (categoryId < 1 || static_cast<std::size_t>(categoryId) > m_Categories.size()) {
+        LOG_ERROR(<< "Programmatic error - invalid ML category: " << categoryId);
+        return 0;
+    }
+    return m_Categories[categoryId - 1].numMatches();
+}
+
+CDataCategorizer::TIntVec CTokenListDataCategorizerBase::usurpedCategories(int categoryId) {
+    CDataCategorizer::TIntVec usurped;
+    if (categoryId < 1 || static_cast<std::size_t>(categoryId) > m_Categories.size()) {
+        LOG_ERROR(<< "Programmatic error - invalid ML category: " << categoryId);
+        return usurped;
+    }
+    auto iter = std::find_if(m_CategoriesByCount.begin(), m_CategoriesByCount.end(), 
+                             [categoryId](const TSizeSizePr& pr){
+                                 return pr.second == static_cast<std::size_t>(categoryId);
+                                 });
+    if (iter == m_CategoriesByCount.end()) {
+        LOG_WARN(<< "Could not find category count for category: " << categoryId);
+        return usurped;
+    }
+    iter++;
```

Comment:
It's best to use `++iter` if you don't care what the old value was.  This is because an iterator can be a non-trivial object and post-increment needs to copy that object so that it can return the old version.  (Maybe this can get optimized away as dead code if the `operator++` is inlined, but still it's better style to use pre-increment than post-increment with non-trivial objects if you don't care about the old value.)